src/Nethermind/Nethermind.Serialization.Rlp/RlpBehaviors.cs

```diff
@@ -3,21 +3,32 @@
 
 using System;
 
-namespace Nethermind.Serialization.Rlp
+namespace Nethermind.Serialization.Rlp;
+
+[Flags]
+public enum RlpBehaviors
 {
-    [Flags]
-    public enum RlpBehaviors
-    {
-        None,
-        AllowExtraBytes = 1,
-        ForSealing = 2,
-        Storage = 4,
-        Eip658Receipts = 8,
-        AllowUnsigned = 16,
-        SkipTypedWrapping = 32, // introduced after typed transactions. In the network (devp2p) transaction has additional wrapping
-                                // when we're calculating tx hash or sending raw transaction we should skip this wrapping
-                                // with additional wrapping for typed transactions we're decoding Uint8Array([TransactionType, TransactionPayload]
-                                // without wrapping we're decoding (TransactionType || TransactionPayload)
-        All = AllowExtraBytes | ForSealing | Storage | Eip658Receipts | AllowUnsigned | SkipTypedWrapping
-    }
+    None,
+    AllowExtraBytes = 1,
+    ForSealing = 2,
+    Storage = 4,
+    Eip658Receipts = 8,
+    AllowUnsigned = 16,
+
+    /// <summary>
+    /// Introduced after typed transactions. In the network (devp2p) transaction has additional wrapping
+    /// when we're calculating tx hash or sending raw transaction we should skip this wrapping
+    /// with additional wrapping for typed transactions we're decoding Uint8Array([TransactionType, TransactionPayload]
+    /// without wrapping we're decoding (TransactionType || TransactionPayload)
+    /// </summary>
+    SkipTypedWrapping = 32,
+
+    /// <summary>
+    /// Transactions like Shard Blob ones can be passed in mempool or execution payload version form
+    /// Execution payload form: TX_TYPE || [chain_id, nonce, ...];
+    /// Mempool form: TX_TYPE || [[chain_id, nonce, ...], &lt;mempool wrapper fields&gt; ].
+    /// See https://eips.ethereum.org/EIPS/eip-4844#networking
+    /// </summary>
+    InMempoolForm = 64,
```

Comment:
Wouldn't it be better to have separated TxDecoders for MemPool form? We are already too deep into parameters.

Reply:
You need to consider wrapping and detect tx type before the decoding, a separated decoding seems to require code duplication with no big profit. Have no better idea for now yet

Reply:
We can try to [#5753 Refactor transaction decoder](https://github.com/NethermindEth/nethermind/issues/5753) in the future